charts/agent-control/charts/agent-control-deployment/templates/uninstall-job.yaml

```diff
@@ -49,6 +50,6 @@ spec:
       
               {{- if include "newrelic-agent-control.auth.secret.shouldRunJob" . }}
               # Delete the secrets created in the cluster
-              kubectl -n {{ $.Release.Namespace }} delete secret {{ include "newrelic-agent-control.auth.secret.name" . }}
+              kubectl -n {{ $.Release.Namespace }} delete secret --ignore-not-found {{ include "newrelic-agent-control.auth.secret.name" . }}
```

Comment:
What about the other resources that we try to delete above? Would it be worth to add the `--ignore-not-found`?

Reply:
The other ones use labels so they don't fail.

Reply:
couldn't be this the secret provided ? I would expect we label any resource we create and then remove only those ones, but not the ones referenced

Reply:
I don't understand what you comment, do you mean to also add a label to the secret? I am just adding this ignore-not-found because this was already created like this.

Reply:
mmm... I see, then we should see if we can label it if not created by the user and only delete it in that case, and since we would be using a label then we also avoid having to add the flag, I will give it a look.

Reply:
I fixed it in last commit.

Reply:
Is this needed? i see that we are rendering this secret, why isn't being remove on uninstall?https://github.com/newrelic/helm-charts/blob/1ad80e482b932c880eabbd730fb3dbfb55c1b5cc/charts/agent-control/charts/agent-control-deployment/templates/secret-sa-auth.yaml#L5

Reply:
perhaps in the past the script was creating the secret and that is why we removed manually? or maybe i'm missing something

Reply:
I believe we are still creating it manually (depending on how values are set up).

The secret is only rendered when it is specified in values

https://github.com/newrelic/helm-charts/blob/36a8f5bf7bfc0ce9aea441537d982f577d1f5a8b/charts/agent-control/charts/agent-control-deployment/templates/_helpers.tpl#L208-L209

Otherwise, it is created manually

https://github.com/newrelic/helm-charts/blob/36a8f5bf7bfc0ce9aea441537d982f577d1f5a8b/charts/agent-control/charts/agent-control-deployment/templates/preinstall-job-register-system-identity.yaml#L281-L287

Reply:
The templated one is only if specified on the values and is either that one or the one by the script https://github.com/newrelic/helm-charts/blob/36a8f5bf7bfc0ce9aea441537d982f577d1f5a8b/charts/agent-control/values.yaml#L166.
So this will only delete the one created by the script.

Reply:
thanks for clarifying this i missed that in some cases the job is creating the secret 👍 , then it make sense to remove it from the unninstall job